examples/pytorch/graphsaint/README.md

```diff
@@ -26,74 +26,120 @@ All datasets used are provided by Author's [code](https://github.com/GraphSAINT/
+python train_sampling.py --task $task
+# e.g. python train_sampling.py --task ppi_n
 ```
 
+- `$task` includes `ppi_n, ppi_e, ppi_rw, flickr_n, flickr_e, flickr_rw, reddit_n, reddit_e, reddit_rw, yelp_n, yelp_e, yelp_rw, amazon_n, amazon_e, amazon_rw`. For example, `ppi_n` represents running experiments on dataset `ppi` with `node sampler`
+
 ## Comparison
 
 * Paper: results from the paper
 * Running: results from experiments with the authors' code
-* DGL: results from experiments with the DGL example
+* DGL: results from experiments with the DGL example. The experiment config comes from `config.py`. You can modify parameters in `config.py` to see different performance of different setup. 
+
+> Note that we implement offline sampling and online sampling in training phase. Offline sampling means all subgraphs utilized in training phase come from pre-sampled subgraphs. Online sampling means we deprecate all pre-sampled subgraphs and re-sample new subgraphs in training phase.
+
+> Note that the sampling method in pre-sampling phase must be offline sampling.
 
 ### F1-micro
 
 #### Random node sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.960±0.001 | 0.507±0.001 |
-| Running | 0.9628 | 0.5077 |
-| DGL | 0.9618 | 0.4828 |
+| Method | PPI | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.960±0.001 | 0.507±0.001 | 0.962±0.001 | 0.641±0.000 | 0.782±0.004 |
+| Running | 0.9628 | 0.5077 | 0.9622 | 0.6393 | 0.7695 |
+| DGL_offline | 0.9715      | 0.4445      | 0.9645 | 0.6457 | 0.8051 |
+| DGL_online | 0.9730 | 0.4358 | 0.9645 | 0.6444 | 0.8014 |
 
 #### Random edge sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.981±0.007 | 0.510±0.002 |
-| Running | 0.9810 | 0.5066 |
-| DGL | 0.9818 | 0.5054 |
+| Method      | PPI         | Flickr      | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.981±0.007 | 0.510±0.002 | 0.966±0.001 | 0.653±0.003 | 0.807±0.001 |
+| Running | 0.9810 | 0.5066 | 0.9656 | 0.6531 | 0.8071 |
+| DGL_offline | 0.9817      | 0.4456      | 0.9655 | 0.6530 | 0.8034 |
+| DGL_online | 0.9815 | 0.4508 | 0.9653 | 0.6516 | 0.7756 |
 
 #### Random walk sampler
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.981±0.004 | 0.511±0.001 |
-| Running | 0.9812 | 0.5104 |
-| DGL | 0.9818 | 0.5018 |
+| Method      | PPI         | Flickr      | Reddit      | Yelp        | Amazon      |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.981±0.004 | 0.511±0.001 | 0.966±0.001 | 0.653±0.003 | 0.815±0.001 |
+| Running | 0.9812 | 0.5104 | 0.9648      | 0.6527      | 0.8131      |
+| DGL_offline | 0.9833      | 0.4592      | 0.9582      | 0.6514      | 0.8178   |
+| DGL_online | 0.9820 | 0.4606 | 0.9572      | 0.6508      | 0.8157   |
 
 ### Sampling time
 
+- Here sampling time includes consumed time of pre-sampling subgraphs and calculating normalization coefficients in the beginning.
```

Comment:
With this definition, I guess we don't need to differentiate the online variant and the offline variant since we always perform offline sampling in the pre-sampling phase, correct?

Reply:
Yes.. Maybe we should discard one of them(offline or online)? Showing only one variant is sufficient.

Reply:
Yes. There will be no need to specify the sampling variant for this table.